internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -48,17 +46,27 @@ var (
 	IgnoreTopologyManagedFieldAnnotation = IgnorePaths{
 		{"metadata", "annotations", clusterv1.ClusterTopologyManagedFieldsAnnotation},
 	}
+	IgnoreNameGenerated = IgnorePaths{
+		{"metadata", "name"},
+	}
 )
 
 func TestReconcileShim(t *testing.T) {
 	infrastructureCluster := builder.InfrastructureCluster(metav1.NamespaceDefault, "infrastructure-cluster1").Build()
 	controlPlane := builder.ControlPlane(metav1.NamespaceDefault, "infrastructure-cluster1").Build()
-	cluster1 := builder.Cluster(metav1.NamespaceDefault, "cluster1").Build()
-	cluster1Shim := clusterShim(cluster1)
+	cluster := builder.Cluster(metav1.NamespaceDefault, "cluster1").Build()
+	cluster.SetUID("foo")
 
 	t.Run("Shim gets created when InfrastructureCluster and ControlPlane object have to be created", func(t *testing.T) {
 		g := NewWithT(t)
 
+		// Create namespace and modify input to have correct namespace set
+		namespace, err := env.CreateNamespace(ctx, "reconcile-cluster-shim")
+		g.Expect(err).To(Succeed())
```

Comment:
nit / just fyi
In general we're using To/ToNot(HaveOccured) for errors and To/ToNot(Succeed) for func calls (similar below)